service/src/main/java/uk/gov/hmcts/reform/fpl/model/Respondent.java

```diff
@@ -32,6 +37,34 @@
     @Builder.Default
     private List<Element<UUID>> representedBy = new ArrayList<>();
 
+    @NotNull(message = "Select if the respondent needs representation")
+    private String legalRepresentation;
+    @Valid
+    private RespondentSolicitor solicitor;
+
+    @JsonIgnore
+    private boolean organisationSpecifiedForRespondentSolicitor;
+
+    @JsonIgnore
+    @AssertTrue(message = "Add organisation details for the respondent solicitor")
+    public boolean isOrganisationSpecifiedForRespondentSolicitor() {
+        if (YES.getValue().equals(legalRepresentation)) {
+            //User selected yes but did not enter any details
+            if (isEmpty(solicitor)) {
+                return false;
+            }
```

Comment:
might it be worth adding a test for this in RespondentTest.java?

Reply:
added tests for this logic